crates/protogen/src/metastore/types/options.rs

```diff
@@ -16,6 +17,48 @@ pub struct InternalColumnDefinition {
     pub arrow_type: DataType,
 }
 
+impl From<InternalColumnDefinition> for OptionValue {
```

Comment:
@scsmithr So i think the serde route may actually be a bit cleaner here. 

The `OptionValue` is already functioning the same as `serde_json::Value`, but with a ton more work on our end. Once we have nested/recursive datatypes that don't translate easily to this structure, such as `DataType`, it requires us to do an opaque serialization anyways, so i think just using serde for everything makes a lot of sense. 

It also adds a TON of boilerplate that is essentially the equivalent of `#[derive(Serialize, Deserialize)]` _(like half of this PR is `From` and `TryFrom` impls that serde would replace)._